app/models/labware_creators/stamped_plate_adding_randomised_controls.rb

```diff
@@ -0,0 +1,255 @@
+# frozen_string_literal: true
+
+require_dependency 'form'
+require_dependency 'labware_creators'
+
+module LabwareCreators
+  # Stamps the majority of samples from the parent plate straight into the child
+  # plate (A1 to A1, B1 to B1 etc.). Also creates and adds a number of randomised
+  # control samples according to configuration from the plate purpose.
+  # NB. This was specifically made for adding 2 controls in Bioscan Lysate plates, with
+  # specific sample metdata, and is fairly specific for that purpose.
+  class StampedPlateAddingRandomisedControls < StampedPlate # rubocop:todo Metrics/ClassLength
+    PARENT_PLATE_INCLUDES =
+      'wells.requests_as_source,wells.requests_as_source.request_type,' \
+        'wells.aliquots,wells.aliquots.sample,wells.aliquots.sample.sample_metadata'
+
+    def parent
+      @parent ||= Sequencescape::Api::V2.plate_with_custom_includes(PARENT_PLATE_INCLUDES, uuid: parent_uuid)
+    end
+
+    # fetch list of controls to add from the child purpose configuration
+    def list_of_controls
+      purpose_config.fetch(:controls)
+    end
+
+    # fetch list of rules from the child purpose configuration
+    def list_of_rules
+      purpose_config.fetch(:control_location_rules)
+    end
+
+    # fetch the project name to use for the control sample from the purpose configuration
+    def control_project_name
+      @control_project_name ||= purpose_config.fetch(:control_project_name)
+    end
+
+    # fetch the study name to use for the control sample from the purpose configuration
+    def control_study_name
+      @control_study_name ||= purpose_config.fetch(:control_study_name)
+    end
+
+    # generate randomised well locations for each control
+    def generate_control_well_locations
+      control_locations = []
+      rules_obeyed = false
+      until rules_obeyed
+        list_of_controls.count.times do |_control_index|
+          # sample a random parent well and fetch its location (child not created yet)
+          location = parent.wells.sample.position['name']
```

Comment:
NB. sample is array.sample() method for random sampling of wells.

Reply:
yes .sample lets you get a random object from the array